### PR TITLE
7903883: --verify-exclude test existence and format checks are broken

### PR DESCRIPTION
**Pull request by LudwikJaniuk**
The way this feature selected "the full set of tests" was actually getting "the set of tests to be run". But often, we run some subset through either specifying test files, folders, or groups. Our problemlist files will still have entries for other tests we were not running right now.

As a solution, --verify-exclude needs to compute the maximal set of tests that could be run for all test suites that any of the tests being run are a part of. I've implemented this by creating a dummy testmanager and passing no parameters to it.

Additionally, the regex for problemlist format had a typo.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [CODETOOLS-7903883](https://bugs.openjdk.org/browse/CODETOOLS-7903883): --verify-exclude test existence and format checks are broken (**Bug** - P4)

### Reviewers
 * [Christian Stein](https://openjdk.org/census#cstein) (@sormuras - Committer) ⚠️ Review applies to [fd5e203f](https://git.openjdk.org/jtreg/pull/236/files/fd5e203f2312194da57146e47ca9edece1daca83)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jtreg.git pull/236/head:pull/236` \
`$ git checkout pull/236`

Update a local copy of the PR: \
`$ git checkout pull/236` \
`$ git pull https://git.openjdk.org/jtreg.git pull/236/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 236`

View PR using the GUI difftool: \
`$ git pr show -t 236`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jtreg/pull/236.diff">https://git.openjdk.org/jtreg/pull/236.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jtreg/pull/236#issuecomment-2473604629)
</details>

### CODE REVIEW

File: src/share/classes/com/sun/javatest/regtest/tool/Tool.java [hunk 1]
```diff
@@ -1178,6 +1180,8 @@ public int run() throws BadArgs, Fault, Harness.Fault, InterruptedException {
         testManager.addTestSpecs(testSpecArgs);
         testManager.addGroupSpecs(testGroupSpecArgs);
 
+        // TODO We know set of suites here, could save/do dummy
+
```
Comment (sormuras) [inline src/share/classes/com/sun/javatest/regtest/tool/Tool.java:1184]:
```suggestion
```

Please don't add `TODO` comments — just do it, or don't.

File: src/share/classes/com/sun/javatest/regtest/tool/Tool.java [hunk 2]
```diff
@@ -1419,25 +1423,37 @@ public int run() throws BadArgs, Fault, Harness.Fault, InterruptedException {
...
+        dummyTestManager.addTestSpecs(testManager.getTestSuites().stream()
+            .map(ts -> TestManager.TestSpec.of(ts.getRootDir().toString()))
+            .collect(Collectors.toList()));
+
         List<String> validTestNames = new ArrayList<String>();
-        List<File> excludeFiles = new ArrayList<File>();
-        for (RegressionTestSuite ts: testManager.getTestSuites()) {
-            RegressionParameters params = createParameters(testManager, ts);
+        for (RegressionTestSuite ts: dummyTestManager.getTestSuites()) {
+            out.println(i18n.getString("main.tests.suite", ts.getRootDir()));
```
Comment (sormuras) [inline src/share/classes/com/sun/javatest/regtest/tool/Tool.java:1436]: Guard this output line by the `--verbose` option?

File: src/share/classes/com/sun/javatest/regtest/tool/Tool.java [hunk 3]
```diff
@@ -1419,25 +1423,37 @@ public int run() throws BadArgs, Fault, Harness.Fault, InterruptedException {
...
+            RegressionParameters params = createParameters(dummyTestManager, ts);
             for (Iterator<TestResult> iter = getResultsIterator(params); iter.hasNext(); ) {
                 TestResult tr = iter.next();
-                out.println(tr.getTestName());
```
Comment (sormuras) [inline src/share/classes/com/sun/javatest/regtest/tool/Tool.java:1429]: This print statement would now produce many output lines by default - perhaps, re-add and guard it via the `--verbose` flag?

**Review by sormuras — CHANGES_REQUESTED** [review]
Looks good so far, left some requests regarding a `TODO` comment and optional verbose output.

**Comment by LudwikJaniuk** [comment]
Thank you for reviewing @sormuras. I agree with all your comments and have addressed them. I chose to put the test names behing -va because that's a huuuuge list for the jdk.

**Review by sormuras — APPROVED** [review]
Looks good to me.

**Review by jaikiran — APPROVED** [review]
I don't have prior knowledge of this code, but reading through what's stated in the linked JBS issue and a brief look at this code, this change appears reasonable to me.

Please generate a jtreg build with this change and use it against the JDK mainline repo once, to make sure this works as expected and doesn't cause any unforeseen failures.

**Comment by jaikiran** [comment]
While at it, I noticed in the linked JBS issue, that the current existing error message says:

> The fully qualified test must exists.

The "exists" should have been "exist". Could you update that error message (and related self reg tests) to fix that typo? If you want to do it in a separate PR, that's fine too.

**Review by jaikiran — APPROVED** [review]

**Comment by LudwikJaniuk** [comment]
I've fixed the exists typo and also ran mainline `configure --with-jtreg=my/build; make test TEST=tier1` with no issues. Therefore integrating.

**Pull request closed without merging**